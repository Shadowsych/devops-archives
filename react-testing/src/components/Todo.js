import React from 'react';

const Todo = (props) => {
    const { todo } = props;
    const todoTitle = <h1>{todo.title}</h1>;
    return (
        <div data-testid={`todo-${todo.id}`}>
            {todo.completed ? <strike>{todoTitle}</strike> : todoTitle}
        </div>
    );
}

export default Todo;
