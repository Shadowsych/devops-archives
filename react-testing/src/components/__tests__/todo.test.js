import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Todo from '../Todo';

test('Todo item with completed false', () => {
    const todo = { id: 1, title: 'Wash dishes', completed: false };
    render(<Todo todo={todo} />);

    const todoElement = screen.getByTestId('todo-1');

    // Verify the element with data-testid "todo-1" to exist in the DOM
    expect(todoElement).toBeInTheDocument();

    // Verify todoElement to have the text "Wash dishes"
    expect(todoElement).toHaveTextContent('Wash dishes');

    // Verify that todoElement is not striked, so that means it's not completed
    expect(todoElement).not.toContainHTML('strike');
});

test('Todo item with completed true', () => {
    const todo = { id: 1, title: 'Make dinner', completed: true };
    render(<Todo todo={todo} />);

    const todoElement = screen.getByTestId('todo-1');

    // Verify the element with data-testid "todo-1" to exist in the DOM
    expect(todoElement).toBeInTheDocument();

    // Verify todoElement to have the text "Make dinner"
    expect(todoElement).toHaveTextContent('Make dinner');

    // Verify that todoElement is striked, so that means it's completed
    expect(todoElement).toContainHTML('strike');
});

test('Renderer tree matches todo.test.js.snap', () => {
    const todo = { id: 1, title: 'Eat food', completed: true };

    // Returns a JSON tree of the Todo component
    const tree = renderer.create(<Todo todo={todo} />).toJSON();

    /**
     * Creates a snapshot todo.test.js.snap and expects the tree to match the snapshot.
     * 
     * Note that if you modify the component, then this will not update the snapshot and
     * will fail the test. If you don't want it to fail, you need to update the snapshot
     * by pressing 'u' after running the tests.
     */
    expect(tree).toMatchSnapshot();
});
