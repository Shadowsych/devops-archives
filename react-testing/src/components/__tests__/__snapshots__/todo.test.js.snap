// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renderer tree matches todo.test.js.snap 1`] = `
<div
  data-testid="todo-1"
>
  <strike>
    <h1>
      Eat food
    </h1>
  </strike>
</div>
`;
